implementation/src/main/java/io/smallrye/jwt/config/JWTAuthContextInfoProvider.java

```diff
@@ -131,16 +133,32 @@ private static JWTAuthContextInfoProvider create(String publicKey, String public
      * HTTP header which is expected to contain a JWT token, default value is 'Authorization'
      */
     @Inject
+    @ConfigProperty(name = "mp.jwt.token.header")
+    private Optional<String> tokenHeader;
+
+    /**
+     * @deprecated Use {@link JWTAuthContextInfoProvider#tokenHeader}
+     */
+    @Inject
     @ConfigProperty(name = "smallrye.jwt.token.header", defaultValue = AUTHORIZATION_HEADER)
-    private String tokenHeader;
+    @Deprecated
+    private String smallRyeTokenHeader;
 
     /**
      * Cookie name containing a JWT token. This property is ignored unless the "smallrye.jwt.token.header" is set to 'Cookie'
      */
     @Inject
-    @ConfigProperty(name = "smallrye.jwt.token.cookie")
+    @ConfigProperty(name = "mp.jwt.token.cookie")
     private Optional<String> tokenCookie;
```

Comment:
Lets have `String` here, @MikeEdgar spotted we do a lot of of Opitonal with the default values, my fault really :-)

Reply:
Actually, when I was looking into this, I was thinking that we should refactor this. I don't think we have any reason to use the injection style to retrieve Configs. We should use the programmatic API. This should simplify the code greatly and allows us to test the different configurations without CDI and without any wrappers.

What do you think?

Reply:
@radcortez this would take things into a completely new direction, lets discuss it separately

Reply:
Yes, it was not to be done in the context of this change, but something to think about or discuss. I can open a separate issue.

Reply:
I think we need the Optionals, so we can issue the log warning messages. If we just set a simple String with the default value, we don't know if we are getting the value set by the user or the default. If we cannot use it without the default either, because it will fail in the injection part.

Reply:
`Optional` makes sense in the absence of a default specified in the config annotation. It's just when optional is used with a default that it becomes unnecessary.

Reply:
Yeah, for this case we do have defaults, but since the config may originate from two different properties, we cannot really use it, because then we won't know if the config was actually set or not to verify the other, unless we check the value to be the default :)

Reply:
@radcortez IMHO we don't need `Optional`, I don't really understand what it is to do with the config originating from the 2 different properties.

Reply:
@radcortez I got it :-), let me remove the other comments

Reply:
Yeah, we will get rid op Optional once we drop eventually the newly deprecated properties